src/dag-node-factory.js

```diff
@@ -0,0 +1,193 @@
+'use strict'
+
+const stable = require('stable')
+const sort = stable.inplace
+// const mh = require('multihashes')
+const protobuf = require('protocol-buffers')
+const proto = protobuf(require('./dag.proto'))
+const util = require('./util')
+const DAGNode = require('./dag-node')
+const DAGLink = require('./dag-link')
+
+class DAGNodeFactory {
+  static create (data, dagLinks, hashAlg, callback) {
```

Comment:
to ref: https://github.com/ipld/js-ipld-dag-pb/pull/7#issuecomment-260371998

> Note: It actualy might make sense to support also a promises API for the .create function since it is a top level consumed object (from ipfs.object.get), so that people don't have their promise chain broken, however, we are not dropping callbacks.